lib/cli/commands/subscribeorders.ts

```diff
@@ -15,20 +15,56 @@ export const builder = {
 };
 
 export const handler = (argv: Arguments) => {
+  ensureConnection(argv, true);
+};
+
+const ensureConnection = (argv: Arguments, printError?: boolean) => {
+  loadXudClient(argv).waitForReady(Number.POSITIVE_INFINITY, (error: Error | null) => {
+    if (error) {
+      if (printError) console.error(`${error.name}: ${error.message}`);
+      setTimeout(ensureConnection.bind(undefined, argv), 3000);
+    } else {
+      console.log('Successfully connected, subscribing for orders');
+      subscribeOrders(argv);
+    }
+  });
+};
+
+const subscribeOrders = (argv: Arguments) =>  {
   const addedOrdersRequest = new xudrpc.SubscribeAddedOrdersRequest();
   addedOrdersRequest.setExisting(argv.existing);
   const addedOrdersSubscription = loadXudClient(argv).subscribeAddedOrders(addedOrdersRequest);
   addedOrdersSubscription.on('data', (order: xudrpc.Order) => {
     console.log(`Order added: ${JSON.stringify(order.toObject())}`);
   });
 
+  // adding end, close, error events only once,
+  // since they'll be thrown for three of subscriptions in the corresponding cases, catching once is enough.
+  addedOrdersSubscription.on('end', reconnect.bind(undefined, argv));
+  addedOrdersSubscription.on('close', reconnect.bind(undefined, argv));
```

Comment:
Based on [this](https://grpc.io/docs/tutorials/basic/node.html#streaming-rpcs) I think we only need to listen to the `end` event and not `close`.

Reply:
call me paranoid, just wanted to be safe with this one, but removing `close`, it's okay.